### PR TITLE
Fix Issue 13663 - Comparison of Tuples with floating point fields

### PR DESCRIPTION
**Pull request by berni44**
This is a copy of #7301. I'm the original author, but as I deleted my old account, I can't do any changes there.

I rebased and replaced "issue 13663" by a link to bugzilla.

### CODE REVIEW

File: std/typecons.d [hunk 1]
```diff
@@ -799,35 +799,73 @@ if (distinctFieldNames!(Specs))
          *           for comparison between `Tuple`s.
          *
          * Returns:
-         * For any values `v1` on the right-hand side and `v2` on the
-         * left-hand side:
+         * For values `v1` on the left-hand side and `v2` on the right-hand side:
+         *
+         * 0 if the expression `v1 == v2` is true for all members, or else for
+         * the first member, where this is not true:
```
Comment (RazvanN7) [inline std/typecons.d:805]: I would rephrase this to:

"For any values `v1` contained by the left-hand side tuple and any values `v2` contained by the right-hand side:

0 if `v1 == v2` for all members or the following value for the first position were the mentioned criteria is not satisfied:"

**Review by RazvanN7 — APPROVED** [review]

File: std/typecons.d (hunk 1 above)
Comment (berni44) [reply std/typecons.d]: done - thanks

File: std/typecons.d [hunk 2]
```diff
@@ -799,35 +799,74 @@ if (distinctFieldNames!(Specs))
          *           for comparison between `Tuple`s.
          *
          * Returns:
-         * For any values `v1` on the right-hand side and `v2` on the
-         * left-hand side:
+         * For any values v1 contained by the left-hand side tuple and any
```
Comment (RazvanN7) [inline std/typecons.d:802]: Sorry for nit picking, but can you use use quotes (`) for expressions: v1, v2, v1 == v2  etc. ?

Comment (berni44) [reply std/typecons.d:802]: Oh, no nit picking, that's fine. I need to find my way into the D conventions again... Will fix this soon.
